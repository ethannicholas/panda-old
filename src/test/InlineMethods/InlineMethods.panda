method test1(a:Array<Int>, fn:(Int)=>(Int)):Array<Int> {
    var result := new Array<Int>(a.length)
    for i, v in a
        result[i] := fn(v)
    return result
}

Console.writeLine(test1([1, 2, 3, 4, 5], function(x:Int):Int {
    return x ^ 2
}))

method test2(a:Array<Int>, fn:(Int, Int)=>(Int)):Int {
    var result := a[0]
    for i in 1 .. a.length
        result := fn(result, a[i])
    return result
}

Console.writeLine(test2([1, 2, 3, 4, 5], function(x:Int, y:Int):Int {
    return x + y
}))
Console.writeLine(test2([1, 2, 3, 4, 5], function(x:Int, y:Int):Int {
    return x * y
}))

var sentence := "I am going to multiply by 3 all 2 numbers in this sentence " +
        "of 16 words"
Console.writeLine(sentence.replace(#/\d+/#, function(number:String):Int {
    return number->>(Int) * 3
}))

method doSomething(m:()=&>()) {
    m()
}

doSomething(method() {
    Console.writeLine("success!")
})