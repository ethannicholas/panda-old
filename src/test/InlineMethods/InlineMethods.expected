[1, 4, 9, 16, 25]
15
120
I am going to multiply by 9 all 6 numbers in this sentence of 48 words
success!
